test/e2e/quick_start_test.go

```diff
@@ -0,0 +1,204 @@
+		Expect(e2eConfig.Variables).To(HaveKey("CONTROL_PLANE_MACHINE_COUNT"), "Invalid configuration. Please add CONTROL_PLANE_MACHINE_COUNT to the e2e.config file")
+		controlPlaneMachineCount, err := strconv.ParseInt(e2eConfig.Variables["CONTROL_PLANE_MACHINE_COUNT"], 10, 0)
+		Expect(err).ToNot(HaveOccurred(), "Invalid configuration. The CONTROL_PLANE_MACHINE_COUNT variable should be a valid integer number")
+		Expect(e2eConfig.Variables).To(HaveKey("WORKER_MACHINE_COUNT"), "Invalid configuration. Please add WORKER_MACHINE_COUNT to the e2e.config file")
+		workerMachineCount, err := strconv.ParseInt(e2eConfig.Variables["WORKER_MACHINE_COUNT"], 10, 0)
+		Expect(err).ToNot(HaveOccurred(), "Invalid configuration. The WORKER_MACHINE_COUNT variable should be a valid integer number")
+
+		By("Running clusterctl config cluster to get a cluster template")
+		workloadClusterTemplate := clusterctl.ConfigCluster(context.TODO(), clusterctl.ConfigClusterInput{
+			// pass reference to the management cluster hosting this test
+			KubeconfigPath: managementClusterKubeConfigPath,
+			// pass the clusterctl config file that points to the local provider repository created for this test,
+			ClusterctlConfigPath: clusterctlConfigPath,
+			// define the cluster template source
+			InfrastructureProvider: e2eConfig.InfraProvider(),
+			// define template variables
+			Namespace:                namespaceName,
+			ClusterName:              clusterName,
+			KubernetesVersion:        e2eConfig.Variables["KUBERNETES_VERSION"],
+			ControlPlaneMachineCount: &controlPlaneMachineCount,
+			WorkerMachineCount:       &workerMachineCount,
+			// setup output path for clusterctl logs
+			LogPath: logPath,
+		})
+		Expect(workloadClusterTemplate).ToNot(BeNil(), "Failed to get the cluster template")
+
+		By("Applying the cluster template to the cluster")
+		managementCluster.Apply(context.TODO(), workloadClusterTemplate)
+
+		By("Waiting for the cluster infrastructure to be provisioned")
+		cluster = discovery.GetClusterByName(context.TODO(), discovery.GetClusterByNameInput{
+			Getter:    client,
+			Name:      clusterName,
+			Namespace: namespaceName,
+		})
+		Expect(cluster).ToNot(BeNil(), "Failed to get the Cluster object")
+		framework.WaitForClusterToProvision(context.TODO(), framework.WaitForClusterToProvisionInput{
+			Getter:  client,
+			Cluster: cluster,
+		}, e2eConfig.IntervalsOrDefault("quick-start/wait-provision-cluster", "3m", "10s")...)
+
+		By("Waiting for the first control plane machine to be provisioned")
+		controlPlane := discovery.GetKubeadmControlPlaneByCluster(context.TODO(), discovery.GetKubeadmControlPlaneByClusterInput{
+			Lister:      client,
+			ClusterName: clusterName,
+			Namespace:   namespaceName,
+		})
+		Expect(controlPlane).ToNot(BeNil())
+		framework.WaitForOneKubeadmControlPlaneMachineToExist(context.TODO(), framework.WaitForOneKubeadmControlPlaneMachineToExistInput{
```

Comment:
It might be better to wait for all machines to be created before the CNI

Reply:
Is there a particular reason to do so?
AFAIK installing as soon as the control plane is up is legit; it is also a precondition to get the control plane ready...